src/renderer/components/SetupTab/ArgInput/index.jsx

```diff
@@ -125,15 +125,26 @@ export default class ArgInput extends React.PureComponent {
     const className = enabled ? null : 'arg-disable';
 
     // These types need a text input, and some also need a file browse button
-    if (
-      [
-        'csv', 'vector', 'raster', 'directory',
-        'freestyle_string', 'number',
-      ].includes(argSpec.type)
-    ) {
-      const typeLabel = argSpec.type === 'freestyle_string'
-        ? 'string'
-        : argSpec.type;
+    if ([
+      'csv', 'vector', 'raster', 'directory', 'file',
+      'freestyle_string', 'number', 'integer', 'percent', 'ratio'
+    ].includes(argSpec.type)) {
+      // These types benefit from more descriptive placeholder text.
+      let placeholderText;
+      switch (argSpec.type) {
+        case 'freestyle_string':
+          placeholderText = 'string';
+          break;
+        case 'percent':
+          placeholderText = 'percent: a number from 0 - 100';
+          break;
+        case 'ratio':
+          placeholderText = 'ratio: a decimal from 0 - 1';
```

Comment:
If it were me I'd just put `ratio (0 - 1)` and `percent (0 - 100)`, but I don't object to this.

Reply:
Either way is probably good. I wanted to deter people from entering fractions, but that's pretty unlikely.